### PR TITLE
Feature/mx1589 dagster job for running all extractors

### PR DESCRIPTION
**Pull request by rababerladuseladim**
# PR Context

Execute the new entrypoint: `pdm run all-extractors`

It is currently broke due to broken extration of rdmo, confluence_vvt and datscha_web...

To test execution of all extractors, I had to refactor all `mocked_extractors` into dedicated pytest plugins. I also unified the `mocked_ldap` fixture across all tests.

# Added

- entrypoint `all-extractors`: run all extractors

### CODE REVIEW

File: tests/confluence_vvt/mocked_confluence_vvt.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from unittest.mock import MagicMock, Mock
+
+import pytest
+import requests
+from _pytest.monkeypatch import MonkeyPatch
```
Comment (cutoffthetop) [inline tests/confluence_vvt/mocked_confluence_vvt.py:5]:
```suggestion
from pytest import MonkeyPatch
```

File: tests/datscha_web/mocked_datscha_web.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
+from typing import Any
+from unittest.mock import MagicMock
+
+import pytest
+from _pytest.monkeypatch import MonkeyPatch
```
Comment (cutoffthetop) [inline tests/datscha_web/mocked_datscha_web.py:5]:
```suggestion
from pytest import MonkeyPatch
```

File: tests/grippeweb/mocked_grippeweb.py [hunk 3]
```diff
@@ -1,17 +1,33 @@
...
+from _pytest.monkeypatch import MonkeyPatch
 from pydantic import BaseModel
-from pytest import MonkeyPatch
 
 from mex.grippeweb.connector import GrippewebConnector
 
-ModelT = TypeVar("ModelT", bound=BaseModel)
+
+@pytest.fixture
+def mocked_grippeweb(mocked_grippeweb_sql_tables, monkeypatch: MonkeyPatch) -> None:
```
Comment (cutoffthetop) [inline tests/grippeweb/mocked_grippeweb.py:11]:
```suggestion
def mocked_grippeweb(
    mocked_grippeweb_sql_tables: dict[str, dict[str, Any]], monkeypatch: MonkeyPatch
) -> None:
```

File: tests/ifsg/mocked_ifsg.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        "parse_rows",
+        lambda self, model: mocked_ifsg_sql_tables[model],
+    )
+
+
+ModelT = TypeVar("ModelT", bound=BaseModel)
+
+
+@pytest.fixture
+def mocked_ifsg_sql_tables() -> dict[ModelT, list[BaseModel]]:
```
Comment (cutoffthetop) [inline tests/ifsg/mocked_ifsg.py:42]:
```suggestion
@pytest.fixture
def mocked_ifsg_sql_tables() -> dict[type[BaseModel], list[dict[str, Any]]:
```

File: tests/ifsg/mocked_ifsg.py [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+from mex.ifsg.models.meta_disease import MetaDisease
+from mex.ifsg.models.meta_field import MetaField
+from mex.ifsg.models.meta_item import MetaItem
+from mex.ifsg.models.meta_schema2field import MetaSchema2Field
+from mex.ifsg.models.meta_schema2type import MetaSchema2Type
+from mex.ifsg.models.meta_type import MetaType
+
+
+@pytest.fixture
+def mocked_ifsg(mocked_ifsg_sql_tables, monkeypatch: MonkeyPatch) -> None:
```
Comment (cutoffthetop) [inline tests/ifsg/mocked_ifsg.py:20]:
```suggestion
def mocked_ifsg(
    mocked_ifsg_sql_tables: dict[type[BaseModel], list[dict[str, Any]]],
    monkeypatch: MonkeyPatch,
) -> None:
```

File: tests/grippeweb/mocked_grippeweb.py [hunk 6]
```diff
@@ -1,17 +1,33 @@
...
+    monkeypatch.setattr(
+        GrippewebConnector,
+        "parse_columns_by_column_name",
+        lambda self, model: mocked_grippeweb_sql_tables[model],
+    )
 
 
 @pytest.fixture
-def mocked_sql_tables() -> dict[str, list[BaseModel]]:
+def mocked_grippeweb_sql_tables() -> dict[str, list[BaseModel]]:
```
Comment (cutoffthetop) [inline tests/grippeweb/mocked_grippeweb.py:30]:
```suggestion
def mocked_grippeweb_sql_tables() -> dict[str, dict[str, Any]]:
```

File: tests/rdmo/mocked_rdmo.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
+from unittest.mock import MagicMock
+
+import pytest
+from _pytest.monkeypatch import MonkeyPatch
```
Comment (cutoffthetop) [inline tests/rdmo/mocked_rdmo.py:4]:
```suggestion
from pytest import MonkeyPatch
```

File: tests/seq_repo/mocked_drop_for_seqrepo.py [hunk 8]
```diff
@@ -0,0 +1,34 @@
+import json
+from pathlib import Path
+from unittest.mock import MagicMock
+
+import pytest
+import requests
+from _pytest.monkeypatch import MonkeyPatch
```
Comment (cutoffthetop) [inline tests/seq_repo/mocked_drop_for_seqrepo.py:7]:
```suggestion
from pytest import MonkeyPatch
```

**Review by cutoffthetop — CHANGES_REQUESTED** [review]
very nice! i like the cleanup of the ldap mocks 🫶 

just a few cosmetic change requests

File: tests/blueant/mocked_blueant.py [hunk 9]
```diff
@@ -0,0 +1,77 @@
+from unittest.mock import MagicMock
+
+import pytest
+import requests
+from _pytest.monkeypatch import MonkeyPatch
```
[the remark this answers is not in the document]
Comment (rababerladuseladim) [reply tests/blueant/mocked_blueant.py]: fixed in cc29b5570bc096bc070e965dbf236534bbf3068f

File: tests/datscha_web/mocked_datscha_web.py (hunk 2 above)
Comment (rababerladuseladim) [reply tests/datscha_web/mocked_datscha_web.py]: fixed in cc29b5570bc096bc070e965dbf236534bbf3068f

File: tests/rdmo/mocked_rdmo.py (hunk 7 above)
Comment (rababerladuseladim) [reply tests/rdmo/mocked_rdmo.py]: fixed in cc29b5570bc096bc070e965dbf236534bbf3068f

File: tests/seq_repo/mocked_drop_for_seqrepo.py (hunk 8 above)
Comment (rababerladuseladim) [reply tests/seq_repo/mocked_drop_for_seqrepo.py]: fixed in cc29b5570bc096bc070e965dbf236534bbf3068f

File: tests/grippeweb/mocked_grippeweb.py (hunk 3 above)
Comment (rababerladuseladim) [reply tests/grippeweb/mocked_grippeweb.py]: fixed in 79b3b8ae6153e3dde8c577f89abf28530006ffab

File: tests/ifsg/mocked_ifsg.py (hunk 4 above)
Comment (rababerladuseladim) [reply tests/ifsg/mocked_ifsg.py]: fixed in f5ffc2323eacf4040ee08fb2820db5f63bd10c3f

File: tests/ifsg/mocked_ifsg.py (hunk 5 above)
Comment (rababerladuseladim) [reply tests/ifsg/mocked_ifsg.py]: fixed in f5ffc2323eacf4040ee08fb2820db5f63bd10c3f

File: tests/grippeweb/mocked_grippeweb.py (hunk 6 above)
Comment (rababerladuseladim) [reply tests/grippeweb/mocked_grippeweb.py]: fixed in 79b3b8ae6153e3dde8c577f89abf28530006ffab

File: pyproject.toml [hunk 10]
```diff
@@ -43,6 +43,7 @@ optional-dependencies.dev = [
 ]
 
 [project.scripts]
+all-extractors = "mex.main:run"
```
[the remark this answers is not in the document]
Comment (rababerladuseladim) [reply pyproject.toml:46]: I settled for `all-extractors` in 71b895cdcb5f7b680ede99e775f4c49567df982b

File: mex/pipeline/base.py [hunk 11]
```diff
@@ -50,6 +50,12 @@ def load_job_definitions() -> Definitions:
         )
         for job in jobs
     ]
+    jobs.append(
+        define_asset_job(
+            "all_assets",
+            AssetSelection.groups(*[group for group in group_names]).upstream(),
```
[the remark this answers is not in the document]
Comment (rababerladuseladim) [reply mex/pipeline/base.py:56]: 🙈  7ab1ec8b89c54223b64e9156193b7a5c3178c3b4

**Review by cutoffthetop — APPROVED** [review]
~just needs a main update, then~ good to merge :)

**Pull request merged**